docs/reference/buildx_build.md

```diff
@@ -0,0 +1,271 @@
+Start a build
+
+### Aliases
+
+`build`, `b`
+
+### Options
+
+| Name | Description |
+| --- | --- |
+| [`--add-host stringSlice`](https://docs.docker.com/engine/reference/commandline/build/#add-entries-to-container-hosts-file---add-host) | Add a custom host-to-IP mapping (host:ip) |
+| [`--allow stringSlice`](#allow) | Allow extra privileged entitlement, e.g. network.host, security.insecure |
+| [`--build-arg stringArray`](https://docs.docker.com/engine/reference/commandline/build/#set-build-time-variables---build-arg) | Set build-time variables |
+| `--builder string` | Override the configured builder instance |
+| [`--cache-from stringArray`](#cache-from) | External cache sources (eg. user/app:cache, type=local,src=path/to/dir) |
+| [`--cache-to stringArray`](#cache-to) | Cache export destinations (eg. user/app:cache, type=local,dest=path/to/dir) |
+| [`-f`](https://docs.docker.com/engine/reference/commandline/build/#specify-a-dockerfile--f), [`--file string`](https://docs.docker.com/engine/reference/commandline/build/#specify-a-dockerfile--f) | Name of the Dockerfile (Default is 'PATH/Dockerfile') |
+| `--iidfile string` | Write the image ID to the file |
+| `--label stringArray` | Set metadata for an image |
+| [`--load`](#load) | Shorthand for --output=type=docker |
+| `--network string` | Set the networking mode for the RUN instructions during build |
+| `--no-cache` | Do not use cache when building the image |
+| [`-o`](#output), [`--output stringArray`](#output) | Output destination (format: type=local,dest=path) |
+| [`--platform stringArray`](#platform) | Set target platform for build |
+| `--progress string` | Set type of progress output (auto, plain, tty). Use plain to show container output |
+| `--pull` | Always attempt to pull a newer version of the image |
+| [`--push`](#push) | Shorthand for --output=type=registry |
+| `--secret stringArray` | Secret file to expose to the build: id=mysecret,src=/local/secret |
+| `--ssh stringArray` | SSH agent socket or keys to expose to the build (format: default|<id>[=<socket>|<key>[,<key>]]) |
+| [`-t`](https://docs.docker.com/engine/reference/commandline/build/#tag-an-image--t), [`--tag stringArray`](https://docs.docker.com/engine/reference/commandline/build/#tag-an-image--t) | Name and optionally a tag in the 'name:tag' format |
+| [`--target string`](https://docs.docker.com/engine/reference/commandline/build/#specifying-target-build-stage---target) | Set the target build stage to build. |
+
+
+<!---MARKER_GEN_END-->
+
+## Description
+
+The `buildx build` command starts a build using BuildKit. This command is similar
+to the UI of `docker build` command and takes the same flags and arguments.
+
+For documentation on most of these flags refer to the [`docker build`
+documentation](https://docs.docker.com/engine/reference/commandline/build/). In
+here we’ll document a subset of the new flags.
+
+## Examples
+
+### <a name="platform"></a> Set the target platforms for the build (--platform)
```

Comment:
Let me try updating my docs PR, and see if it likes this, or if it expects the "extended" syntax; https://www.markdownguide.org/extended-syntax/#heading-ids

```suggestion
### Set the target platforms for the build (--platform) {#platform}
```

Reply:
If they are functionally equivalent then we can use the markdown style. Need to update the generator as well as it is looking for HTML style marker atm. to see if link is possible.

Reply:
Ok, I tried manually updating to the markdown syntax in https://github.com/docker/docker.github.io/pull/12092 (preview  at https://deploy-preview-12092--docsdocker.netlify.app/engine/reference/commandline/buildx_build/

So both work, but looks like the Markdown syntax works better, and is cleaner; in that case Jekyll accepts the custom ID; skipping its own generated one. So in that case there won't be two IDs (one custom and one generated);

Before:

<img width="664" alt="Screenshot 2021-03-24 at 16 14 43" src="https://user-images.githubusercontent.com/1804568/112339869-573a9a80-8cc0-11eb-8b99-14ce3a00b634.png">


After (with Markdown syntax):

<img width="473" alt="Screenshot 2021-03-24 at 16 41 23" src="https://user-images.githubusercontent.com/1804568/112339895-5c97e500-8cc0-11eb-908e-b1841c12e157.png">


I haven't tested what it looks like on GitHub though. Let's try if it works in this comment

### Example with custom anchor-id  {#mycustomid}

this is a link [to the custom ID](#mycustomid)


Sigh... doesn't seem to work in GFM 😞 (I hate all the different flavours of markdown); https://github.github.com/gfm/

Let me dig a bit more to see if it's possible, otherwise we use the current approach

Reply:
Looks like it's not possible with GitHub's markdown flavour, so let's keep the markdown as-is; perhaps I'll change the yamldocs generator to change it, but that's not for this PR